cpp/src/io/csv/type_conversion.cuh

```diff
@@ -21,17 +21,15 @@ bool isDigit(char data) {
 
 
 __host__ __device__
-void adjustForWhitespaceAndQuotes(const char *data, long& start_idx, long& end_idx, char quotechar='\0') {
```

Comment:
@thomcom Do you know why this did not work as expected? Is it a quirk of nvcc, since the funciton is marked as __host__ __device__?

Reply:
I was wondering the same thing.

Reply:
Weird.

Reply:
I find pointers clearer than references for non-const parameters anyway.

Reply:
I would really like to make sure that this change is required, because it negatively affects the function readability IMO.

Reply:
> I find pointers clearer than references for non-const parameters anyway.
I guess we disagree on this one :)

Reply:
If and when we get a code style guide made, one of the first things I'll be advocating for is requiring that passing parameters by reference should only be done for input parameters, not output parameters. This is inline with the Google C++ guide, which we kind of follow.

Reply:
I do hope we get a style guide sooner rather than later. I don't mind either way as long as there's consistency. FWIW, the IsoCpp guidelines prefer using pass by non-const reference for in-out parameters.

Reply:
Why is this is `__host__ __device__` function instead of just `__device__`? It's only used in kernels. 

You can't pass by ref to `__global__` functions memory that isn't UVM. But that's not happening here.

There should be no problem passing by reference into `__device__` functions.

Reply:
@vuule this is a good point -- do you need this function to be called from the __host__?

Reply:
I locally reverted this part of the change and all tests are still passing. I does not look like this change was needed. 
But, if this is the style we want to use, it's fine anyhow.

Also, yes, this function can be __device__ only.

Reply:
The change is needed - the test is bad. In particular it mustn't have the types in the buffer string, or type inference will not be used.

I reverted to `bfd425a5fca16ee` and added the following code to `test_csvreader.py` to validate that the fix is necessary:

```
def test_csv_reader_dtype_inference_whitespace():
    lines = ['  1.2,    1',
             '2.3,2    ',
             '  3.4,   3',
             ' 4.5,4',
             '5.6,  5',
             ' 6.7,6 ']
    buffer = '\n'.join(lines) + '\n'
    cu_df = read_csv(StringIO(buffer), compression=None)
    pd_df = pd.read_csv(StringIO(buffer))
    print(cu_df)
    print(pd_df)
    assert False
```
output:
```
E       assert False

cudf/tests/test_csvreader.py:466: AssertionError
----------------------------- Captured stdout call -----------------------------
         1.2          1
0   79776054   20819818
1 1668713040  728791489
2  -92221750 1905524192
3   79776054   79776054
4  -92221750   -3501432
     1.2      1
0    2.3      2
1    3.4      3
2    4.5      4
3    5.6      5
4    6.7      6
```

Reply:
I still don't know exactly why the change was required but it was good that it got merged. :)